src/storage/table/column_segment.cpp

```diff
@@ -9,10 +9,62 @@ ColumnSegment::ColumnSegment(TypeId type, ColumnSegmentType segment_type, idx_t
       stats(type, type_size) {
 }
 
+ColumnSegment::ColumnSegment(TypeId type, ColumnSegmentType segment_type, idx_t start, idx_t count,data_t&  stats_min, data_t&  stats_max)
+    : SegmentBase(start, count), type(type), type_size(GetTypeIdSize(type)), segment_type(segment_type),
+      stats(type, type_size,stats_min,stats_max) {
+}
+
 SegmentStatistics::SegmentStatistics(TypeId type, idx_t type_size) : type(type), type_size(type_size) {
 	Reset();
 }
 
+template <class T> static void set_min_max(data_t& min_value_p, data_t& max_value_p,  data_t& min_p, data_t& max_p) {
+    memcpy(&min_p,&min_value_p, sizeof(T));
+    memcpy(&max_p,&max_value_p, sizeof(T));
+
+//    auto min = (T *)min_p;
+//    auto max = (T *)max_p;
+//    auto min_value = (T *)min_value_p;
+//    auto max_value = (T *)max_value_p;
+//    *min = *min_value;
+//    *max = *max_value;
+}
+
+
+
+SegmentStatistics::SegmentStatistics(TypeId type, idx_t type_size, data_t& stats_min, data_t&  stats_max) : type(type), type_size(type_size){
```

Comment:
Could you change these parameters to `data_t stats_min[], data_t stats_max[]`? The way it is handled right now is using a reference to put in the first byte, and then using that is a pointer, which is a bit counter intuitive.